app/src/main/java/org/mozilla/focus/widget/Cursor.java

```diff
@@ -79,15 +79,19 @@ public void stopMoving(Direction direction) {
         if (activeDirections.size() == 0) {
             handler.removeCallbacks(tick);
             moving = false;
+            speed = 0;
         }
     }
 
     private void move() {
+        speed++;
+        speed *= FRICTION;
```

Comment:
nit: I'm not sure this is most natural way to write this – it seems strange to modify the velocity by 1 and then multiply the whole velocity by a constant. Typically, I think you'd achieve this affect by modifying the acceleration (the 1 value you're adding) instead.

That being said, I'm not sure if ^ that's just more poor understanding of physics and how you'd come up with a better solution anyway.